### PR TITLE
stackItems attempt 2

### PR DESCRIPTION
**Pull request by acemac**
**Issue Number**

#397 replacement because I'm terrible at using git

**Overview of PR**

Removing the API updates that were accidentally merged in to #397 turned out to be an ordeal. This PR should be cleaner.

_Don't forget to update the CHANGELOG.md file with any changes that are in this PR_

### CODE REVIEW

File: src/lib/items/Item.js [hunk 1]
```diff
@@ -102,7 +102,8 @@ export default class Item extends Component {
       nextProps.canvasTimeStart !== this.props.canvasTimeStart ||
       nextProps.canvasTimeEnd !== this.props.canvasTimeEnd ||
       nextProps.canvasWidth !== this.props.canvasWidth ||
-      nextProps.order !== this.props.order ||
+      (nextProps.order ? nextProps.order.idnex : undefined) !==
```
[the remark this answers is not in the document]
Comment (acemac) [reply src/lib/items/Item.js]: Good catch. I clearly didn't give this PR the love it deserves.

**Comment by acemac** [comment]
Thank you for reviewing.

Most of `calendar.js` that I change I don't fully understand the implementation, I'm just doing copy paste job on functions and changing where required. 

I would prefer to open a new issue for coverage and testing. Code coverage of this component is around 23% at the moment. There is very little testing for the user interaction, which is what this component is built around. I currently have no way to test #401

**Comment by Ilaiwi** [comment]
@acemac  I found this line 
https://github.com/namespace-ee/react-calendar-timeline/blob/master/src/lib/utility/calendar.js#L444
any idea on what is `isOverlay`?

**Comment by Ilaiwi** [comment]
@acemac this code in `stackItems` is like a boom. I will get to unit test it after `custom headers` hopefully

**Comment by Ilaiwi** [comment]
@acemac I've been unit testing and refactoring `stackItems` and `stackAll` utils from this PR. It is still work in progress but I would appreciate your feedback
https://github.com/acemac/react-calendar-timeline/pull/2

**Pull request merged**